### ISSUE TITLE
Double RSC fetch after server action redirect

### ISSUE THREAD

**Issue by cfrank** [vercel/next.js#66166]
### Link to the code that reproduces this issue

https://github.com/cfrank/next.js

### To Reproduce

1. Clone NextJS canary
2. `pnpm i`
3. `pnpm build`
4. `NEXT_TEST_SKIP_CLEANUP=1 pnpm test-start test/e2e/app-dir/actions/app-action.test.ts`
5. Visit temporary next install directory (`/tmp/next-install-*`)
6. Run `pnpm next`
7. Visit [http://localhost:3000/client](http://localhost:3000/client)
8. Click "redirect"

### Current vs. Expected behavior

Currently when performing a redirect within a server action we are correctly resolving the RSC bundle within the `POST` request and performing the client side redirect to the destination path. But upon loading the destination path we are performing a subsequent fetch of the destination path:

![image](https://github.com/vercel/next.js/assets/2308484/be06f08b-cc4e-425f-8553-85088cb01bec)

### Provide environment information

```bash
Operating System:
  Platform: Linux
  Arch: amd64
  Version: `6.8.0-31-generic #31-Ubuntu SMP PREEMPT_DYNAMIC Sat Apr 20 00:40:06 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux`

Binaries:
  Node: `v21.7.3`
  NPM: `10.5.0`
  pnpm: `8.15.7`

Relevant Packages:
  NextJS: `v14.3.0-canary.80`
```

### Which area(s) are affected? (Select all that apply)

Navigation

### Which stage(s) are affected? (Select all that apply)

next dev (local), next start (local), Other (Deployed)

### Additional context

I encountered this bug while working on https://github.com/vercel/next.js/pull/64604 where I added more robust tests to verify that only a single request was being made during server action redirects. When I opened the PR the tests were passing and only a single POST request was being made with all relevant data being shipped within that single request, but shortly after a change was landed which introduced this second request.

**Comment by icyJoseph** [comment]
This has been reported here as well:

- https://github.com/vercel/next.js/issues/65764

Since the _rsc request is discarded, in light of a redirect, the tests setup in test/e2e/app-dir/navigation/navigation.test.ts pass, but the issue is there too.

For example, say, you have this route handler:

```js
let t = 0
const getTime = () => t++

export const GET = (req) => {
  const rsc = req.nextUrl.searchParams.get('_rsc')
  return new Response(
    `<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="UTF-8">
        <title>Document</title>
    </head>

    <body>
        <h1 id="result-page">Result Page</h1>
        <div id="counter">${getTime()}</div>
        <div id="rsc">${rsc || 'no-rsc'}</div>
    </body>
</html>
  `,
    { status: 200, statusText: 'OK', headers: { 'Content-Type': 'text/html' } }
  )
}

```

Then you can write a test:

```js
   it('should only trigger the redirect once for Route Handlers', async () => {
        const path = '/redirect/redirect-with-loading/to-api'
        const browser = await next.browser(path)

        const counter = await browser.waitForElementByCss('#counter').text()

        expect(counter).toBe('0')
      })
```

And that'll error out because counter will be actually 1.

I haven't had time to deep dive into why the _rsc request is made though

**Comment by cfrank** [comment]
Fixed #66620
